receiver/vcenterreceiver/client.go

```diff
@@ -91,6 +106,7 @@ func (vc *vcenterClient) Datacenters(ctx context.Context) ([]mo.Datacenter, erro
 	if err != nil {
 		return nil, fmt.Errorf("unable to retrieve Datacenters: %w", err)
 	}
+	defer func() { _ = v.Destroy(ctx) }()
```

Comment:
Is this an unrelated bugfix? If so, can we have a separate PR for it?

Reply:
@djaglowski have a separate PR here https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/34254